lib/upgrade/migrator/migrate_test.go

```diff
@@ -57,3 +63,150 @@ var _ = Describe("UT for checking the version for migration.", func() {
 		Entry("Expect 1.2.3.4.5 to not migrate", "1.2.3.4.5", false, true),
 	)
 })
+
+type singleTypeClient struct {
```

Comment:
Why is `singleTypeClient` needed for the Profile tests (given that `fakeClientV1` apparently works fine for the WEP tests)?

Reply:
The reason it does not work is because the fakeClientV1 looks through the Keys to see if they match the ListOptions Key to generate a list of the KV pairs.  To do this it takes the Keys and does a KeyToDefaultPath which for a model.ProfileKey produces `/calico/v1/policy/profile/profileName` which is, unfortunately, not matched by the Profile match regex https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/model/profile.go#L30.  I'm willing to switch back to the fakeClientV1 if you can help me make it work, I did spend some time trying to use it.

Reply:
Hmm, it feels to me that something is wrong here.  AFAIU, Profile is just as valid as WorkloadEndpoint, in both v1 and v3 data models - so any approach that works for WorkloadEndpoint should work for Profile as well.  I suggest checking this with Rob.